### PR TITLE
Introduce Span component, properly apply color

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- Added `Span` - same as `Text` without the annoying `fontSize="medium"` (inherits by default instead)
  - Replace all library-internal usage of `Text` with `Span`
- `Code`, `CodeBlock` & `Paragraph` now explicitly use `theme.colors.text` as default color
- Added test coverage for `Span` using image-snapshots
- Simplified test coverage for `Text` to verify that default fontSize is properly assigned.
- Replaced in-library usage of `Text` with `Span`
- Updated documentation to reflect preference for `Span` component where possible

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## UNRELEASED
 
+<<<<<<< HEAD
```
Comment (jhardy) [inline CHANGELOG.md:10]:
```suggestion
```

File: www/src/documentation/components/text/text.mdx [hunk 2]
```diff
@@ -1,14 +1,16 @@
 ---
```
Comment (jhardy) [inline www/src/documentation/components/text/text.mdx:1]: image

Span is showing up as `Text` in the sidebar. Maybe that is ok?

File: packages/components/src/Text/__snapshots__/CodeBlock.test.tsx.snap [hunk 3]
```diff
@@ -6,12 +6,12 @@ exports[`A default CodeBlock component 1`] = `
   font-size: 0.875rem;
   line-height: 1.25rem;
   padding: 1rem;
-  color: #262D33;
 }
 
 .c1 {
   border: 1px solid;
   border-color: #DEE1E5;
+  color: #262D33;
```
Comment (jhardy) [inline packages/components/src/Text/__snapshots__/CodeBlock.test.tsx.snap:14]: Should `CodeBlock` have a color set on it?

**Review by jhardy — CHANGES_REQUESTED** [review]
Overall looks good, just a few small questions and one fix for the changelog

File: www/src/documentation/components/text/text.mdx (hunk 2 above)
Comment (lukelooker) [reply www/src/documentation/components/text/text.mdx:1]: Ack. Nah - we should be consistent (boy do I look forward to landing new Gatsby setup that lets us stop managing this manually )

File: packages/components/src/Text/__snapshots__/CodeBlock.test.tsx.snap (hunk 3 above)
Comment (lukelooker) [reply packages/components/src/Text/__snapshots__/CodeBlock.test.tsx.snap:14]: Yes, that was intentional (it was actually the original issue that came from API Explorer)

**Comment by lukelooker** [comment]
> Overall looks good, just a few small questions and one fix for the changelog

Feedback addressed! Thank you :)

**Review by jhardy — APPROVED** [review]

**Pull request merged**